### PR TITLE
`[SK2]`: Fix syncPurchases

### PR DESCRIPTION
**Pull request by MarkVillacampa**
We can use any transaction token available, not only auto-renewable ones

### CODE REVIEW

File: Sources/Purchasing/Purchases/PurchasesOrchestrator.swift [hunk 1]
```diff
@@ -1307,7 +1307,7 @@ private extension PurchasesOrchestrator {
                              completion: @escaping @Sendable (Result<PromotionalOffer, PurchasesError>) -> Void) {
 
         _ = Task<Void, Never> {
-            let transaction = await self.transactionFetcher.firstVerifiedAutoRenewableTransaction
+            let transaction = await self.transactionFetcher.firstVerifiedTransaction
```
Comment (tonidero) [inline Sources/Purchasing/Purchases/PurchasesOrchestrator.swift:1310]: Should this only be subscriptions? Not sure if we can have promotional purchases for non-subscriptions?

**Review by joshdholtz — APPROVED** [review]
:shipit:

**Review by tonidero — APPROVED** [review]
Left a comment about the promotionalOffers but should be ok for the syncPurchases I think

File: Sources/Purchasing/Purchases/PurchasesOrchestrator.swift (hunk 1 above)
Comment (MarkVillacampa) [reply Sources/Purchasing/Purchases/PurchasesOrchestrator.swift]: Changed back to return only subscriptions for promotional offers 👍

**Pull request merged**